src/components/GameLife/GameLife.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import _ from 'lodash';
+import { css } from '@emotion/core';
+
+import { FieldSizeEnum, StartPercentEnum, AgeEnum, SpeedEnum, OperationEnum } from '@shared/enums';
+import { randomInteger } from '@shared/utils';
+import { Cell, CellInterface } from './components/Cell';
+import { BlockSize, HandleFieldSize } from './components/BlockSize';
+import { BlockStartPercent, HandleStartPercent } from './components/BlockStartPercent';
+import { Pult, HandlePult } from './components/Pult';
+import { Button } from './components/components/Button';
+
+type FieldSize = FieldSizeEnum.Small | FieldSizeEnum.Medium | FieldSizeEnum.Big;
+type StartPercent = StartPercentEnum.Small | StartPercentEnum.Medium | StartPercentEnum.Big;
+type Speed = SpeedEnum.Small | SpeedEnum.Medium | SpeedEnum.Big;
+type RandomStoreType = { [key: number]: boolean };
+
+interface GameLifeProps {};
+interface GameLifeState {
+    data: CellInterface[];
+    fieldSize: FieldSize;
+    startPercent: StartPercent,
+    speed: Speed,
+    active: boolean
+};
+
+export class GameLife extends React.PureComponent<GameLifeProps, GameLifeState> {
+    constructor(props: GameLifeProps) {
+        super(props);
+
+        this.state = {
+            // data: this.createData(FieldSizeEnum.Small, StartPercentEnum.Medium), HW 30.04.2020
+            data: this.createData(FieldSizeEnum.Small),
+            fieldSize: FieldSizeEnum.Small,
+            startPercent: StartPercentEnum.Medium,
+            speed: SpeedEnum.Small,
+            active: false
+        };
+    }
+
+    generateRandomNumbers = (fieldSize:FieldSize, startPercent:StartPercent): RandomStoreType => {
+        const store: RandomStoreType = {};
+        const max = fieldSize * (fieldSize - 20) - 1;
+        const storeLength = Math.round(startPercent * max / 100);
+
+        while (Object.keys(store).length < storeLength) {
+            const num: number = randomInteger(0, max);
+
+            if (!store[num]) {
+                store[num] = true;
+            }
+        }
+
+        return store;
+    }
+
+    createData = (l: FieldSize):CellInterface[] => {
+    // createData = (l: FieldSize, p: StartPercent):CellInterface[] => { HW 30.04.2020
+        const dataLength: number = l * (l - 20);
+        // const randomNumbers: RandomStoreType = this.generateRandomNumbers(l, p); HW 30.04.2020
+        
+        return [...new Array(dataLength)].map((_, index): CellInterface => {
```

Comment:
```suggestion
        return [...new Array(dataLength)].map((_, position): CellInterface => {
```

называй переменные как удобно

Reply:
и ниже `return { position, age }`

Reply:
\+